kernel/src/vm/vmar/mod.rs

```diff
@@ -49,11 +49,11 @@ pub struct Vmar {
 
 impl Vmar {
     /// Creates a new VMAR.
-    pub fn new() -> Arc<Self> {
+    pub fn new(executable_file: PathOrInode) -> Arc<Self> {
```

Comment:
I think it is slightly better to accept a `ProcessVm` instance here.

Maybe `ProcessVm::new` can be made private after that.